sycl/test-e2e/no-xfail-without-tracker.cpp

```diff
@@ -19,15 +19,19 @@
 // - in the result, search for "XFAIL" again, but invert the result - this
 //   allows us to get the line *after* XFAIL
 // - in those lines, check that XFAIL-TRACKER is present and correct. Once
-//   again, invert the search to get all "bad" lines
+//   again, invert the search to get all "bad" lines and save the test names in
+//   improperly-xfailed-tests.txt
 // - make a final count of how many ill-formatted directives there are and
 //   verify that against the reference
+// - ...and check if the list of improperly "xfailed" tests needs to be updated.
 //
-// RUN: grep -rI "XFAIL:" %S -A 1 --include=*.c --include=*.cpp \
-// RUN:     --no-group-separator | \
+// RUN: grep -rI "XFAIL:" %S -A 1 --exclude=%s --include=*.c --include=*.cpp \
+// RUN:     --exclude=no-xfail-without-tracker.cpp --no-group-separator | \
 // RUN: grep -v "XFAIL:" | \
-// RUN: grep -Pv "XFAIL-TRACKER:\s+(?:https://github.com/[\w\d-]+/[\w\d-]+/issues/[\d]+)|(?:[\w]+-[\d]+)" | \
-// RUN: wc -l | FileCheck %s --check-prefix NUMBER-OF-XFAIL-WITHOUT-TRACKER
+// RUN: grep -Pv "XFAIL-TRACKER:\s+(?:https://github.com/[\w\d-]+/[\w\d-]+/issues/[\d]+)|(?:[\w]+-[\d]+)" > improperly-xfailed-tests.txt | \
```

Comment:
We should use `%t` for temporary files, otherwise we risk that files will be created in a wrong location, polluting source tree, or not being removed later